cpp/arcticdb/storage/storage.hpp

```diff
@@ -91,6 +106,14 @@ class Storage {
     Storage(Storage&&) = default;
     Storage& operator=(Storage&&) = delete;
 
+    /**
+     * Check if (at least) read access is possible on this Storage and return the check result.
+     * Ideally should not throw unless the function is certain no subsequent call on this Storage will ever work.
+     *
+     * Optional feature. Only implement if the check is cheap and fast (returns within 2s).
```

Comment:
What does optional mean? This is pure virtual function so it will be implemented by everyone down the inheritance chain.

Reply:
Optional means an implementation doesn't have to do any actual check.

Reply:
I'll reword in the next push